pkg/controllers/disruption/consolidation.go

```diff
@@ -177,7 +182,39 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 
 	if allExistingAreSpot &&
 		results.NewNodeClaims[0].Requirements.Get(v1beta1.CapacityTypeLabelKey).Has(v1beta1.CapacityTypeSpot) {
-		if len(candidates) == 1 {
+		// Allow spot consolidation if SpotConsolidation featureflag is set to true by customers
+		if options.FromContext(ctx).FeatureGates.SpotConsolidation {
+			var allowConsolidation bool = true
+			// All possible replacements for the current candidate sorted by price
+			instanceTypesOrderedByPrice := results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions.OrderByPrice(results.NewNodeClaims[0].Requirements)
+
+			// Allow single-node/multi-node consolidation only if there are minimum number of possible InstanceTypes for a NodeClaim
+			if len(instanceTypesOrderedByPrice) >= MinInstanceTypesForSpotConsolidation {
+				// For single-node consolidation, also check if the current candidate is part of the window to avoid consolidation
+				if len(candidates) == 1 {
+					currentCandidateName := candidates[0].instanceType.Name
+					for _, instanceType := range lo.Slice(instanceTypesOrderedByPrice, 0, MinInstanceTypesForSpotConsolidation) {
```

Comment:
This creates an entirely new slice which is less efficient here, especially given the number of times that we are running this function. We should probably just iterate until we get to the `MinInstanceTypesForSpotConsolidation` item here rather than generating a new slice

Reply:
Makes sense. Addressed this.